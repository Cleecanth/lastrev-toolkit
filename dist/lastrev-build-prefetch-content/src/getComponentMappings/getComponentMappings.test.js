"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var getComponentMappings_1 = __importDefault(require("./getComponentMappings"));
var getComponentMappings_mock_1 = require("./getComponentMappings.mock");
describe('getComponentMappings.js', function () {
    test('returns correct mappings with no overrides or exclude', function () {
        var mappings = getComponentMappings_1.default(getComponentMappings_mock_1.componentNames, getComponentMappings_mock_1.contentTypes);
        expect(mappings).toMatchSnapshot();
    });
    test('returns correct mappings with overrides and no exclude', function () {
        var mappings = getComponentMappings_1.default(getComponentMappings_mock_1.componentNames, getComponentMappings_mock_1.contentTypes, {
            overrides: {
                settingsGlobal: 'Layout'
            }
        });
        expect(mappings).toMatchSnapshot();
    });
    test('returns correct mappings with exclude and no overrides', function () {
        var mappings = getComponentMappings_1.default(getComponentMappings_mock_1.componentNames, getComponentMappings_mock_1.contentTypes, {
            exclude: ['PageGeneral']
        });
        expect(mappings).toMatchSnapshot();
    });
    test('returns correct mappings with both overrides and excludes', function () {
        var mappings = getComponentMappings_1.default(getComponentMappings_mock_1.componentNames, getComponentMappings_mock_1.contentTypes, {
            exclude: ['PageGeneral'],
            overrides: {
                settingsGlobal: 'Layout'
            }
        });
        expect(mappings).toMatchSnapshot();
    });
    test('returns correct mappings with override containing non-existent component', function () {
        var mappings = getComponentMappings_1.default(getComponentMappings_mock_1.componentNames, getComponentMappings_mock_1.contentTypes, {
            overrides: {
                pageGeneral: 'Dummy'
            }
        });
        expect(mappings).toMatchSnapshot();
    });
    test('returns correct mappings with override containing non-existent content type', function () {
        var mappings = getComponentMappings_1.default(getComponentMappings_mock_1.componentNames, getComponentMappings_mock_1.contentTypes, {
            overrides: {
                dummy: 'Dummy'
            }
        });
        expect(mappings).toMatchSnapshot();
    });
    test('returns correct mappings with exclude containing non-existent component', function () {
        var mappings = getComponentMappings_1.default(getComponentMappings_mock_1.componentNames, getComponentMappings_mock_1.contentTypes, {
            overrides: {
                dummy: 'Dummy'
            }
        });
        expect(mappings).toMatchSnapshot();
    });
    test('returns correct mappings with overrides but exclude contains overriden component', function () {
        var mappings = getComponentMappings_1.default(getComponentMappings_mock_1.componentNames, getComponentMappings_mock_1.contentTypes, {
            exclude: ['Layout'],
            overrides: {
                settingsGlobal: 'Layout'
            }
        });
        expect(mappings).toMatchSnapshot();
    });
});
