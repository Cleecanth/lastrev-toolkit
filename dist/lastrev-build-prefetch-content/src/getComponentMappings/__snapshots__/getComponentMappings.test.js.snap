// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getComponentMappings.js returns correct mappings with both overrides and excludes 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "settingsGlobal": "Layout",
}
`;

exports[`getComponentMappings.js returns correct mappings with exclude and no overrides 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
}
`;

exports[`getComponentMappings.js returns correct mappings with exclude containing non-existent component 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "pageGeneral": "PageGeneral",
}
`;

exports[`getComponentMappings.js returns correct mappings with no overrides or exclude 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "pageGeneral": "PageGeneral",
}
`;

exports[`getComponentMappings.js returns correct mappings with override containing non-existent component 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
}
`;

exports[`getComponentMappings.js returns correct mappings with override containing non-existent content type 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "pageGeneral": "PageGeneral",
}
`;

exports[`getComponentMappings.js returns correct mappings with overrides and no exclude 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "pageGeneral": "PageGeneral",
  "settingsGlobal": "Layout",
}
`;

exports[`getComponentMappings.js returns correct mappings with overrides but exclude contains overriden component 1`] = `
Object {
  "moduleCardGeneral": "ModuleCardGeneral",
  "pageGeneral": "PageGeneral",
}
`;
